docs/resource-policy/policy/topology-aware.md

```diff
@@ -445,56 +445,102 @@ metadata:
-is generating topology hints, it will consult this allowed / denied path
-list to determine what hints are created. The deny path list is checked
-first and then the allowed path list. This means that you can deny all
-the mount/device paths and then allow only the needed ones for example.
-User can either set the path with "prefix" (this is the default) or with
-"glob" type. The "prefix" type means that the prefix of the mount/device
-path is checked for matches. The "glob" type means that user is able to
-put wildcards to the matched paths.
+It is possible to control whether and what kind of topology hints are
+generated using extra pod annotations. By default hints are generated
+from mounts and devices injected into the container. If pod resource API
+queries are enabled, query replies are also used for hint generation.
+
+### Enabling Or Disabling Selected Types of Topology Hints
+
+The `topologyhints.resource-policy.nri.io` annotation key can be used
+to enable or disable topology hint generation for one or more containers
+altogether, or selectively for mounts, devices, and pod resources types.
+More than one type can be specified as a comma-separated list. Additionally,
+the `all` and `none` types are recognized to mean all or none of these
+types. If no hint type annotations are present, all types of hints are
+enabled.
+
+For example:
+
+```yaml
+metadata:
+  annotations:
+    # disable topology hint generation for all containers by default
+    topologyhints.resource-policy.nri.io/pod: none
+    # enable mount-based hints for the 'diskwriter' container
+    topologyhints.resource-policy.nri.io/container.diskwriter: mounts
+    # enable device-based hints for the 'videoencoder' container
+    topologyhints.resource-policy.nri.io/container.diskwriter: devices
+    # enable pod resource-based hints for the 'dpdk' container
+    topologyhints.resource-policy.nri.io/container.dpdk: pod-resources
+    # enable device and pod resource-based hints for 'networkpump' container
+    topologyhints.resource-policy.nri.io/container.networkpump: devices,pod-resources
+```
+
+Note that for pod resource based hints, you also need to enable pod
+resource API queries using the corresponding configuration option, like
+this:
+
+```yaml
+apiVersion: config.nri/v1alpha1
+kind: TopologyAwarePolicy
+metadata:
+  name: default
+  ...
+spec:
+  ...
+  agent:
+    podResourceAPI: true
+```
+
+### Controlling Topology Hints by Path
+
+It is also possible to enable and disable topology hint generation based
+on mount or device path, using allow and deny lists. When the policy is
+tenerating topology hints, it consults these lists to decide whether hints
+for a particular mount or device is enabled. The denied list is consulted
+first, followed by the allow list. The common pattern for of usage is to
+deny all paths and then to allow only selected ones.
+
+Two types of list are supported, glob- and prefix-matched lists. Entries
```

Comment:
Suggesting a rewrite of this paragraph.

Two types of deny/allow lists are supported: glob and prefix.
A path matches a prefix list if it starts with any entry in the list.
A path matches a glob list, where entries include shell-style path globbing with wildcards, it the full path matches any glob pattern in the list.